ext/zip/php_zip.c

```diff
@@ -1004,7 +1004,6 @@ static void php_zip_object_free_storage(zend_object *object) /* {{{ */
 	if (intern->za) {
 		if (zip_close(intern->za) != 0) {
 			php_error_docref(NULL, E_WARNING, "Cannot destroy the zip context: %s", zip_strerror(intern->za));
-			return;
```

Comment:
`intern->za` was not NULL, so it's probably not destroyed at this point. In further, there are also two superfluous NULL assignments.

Thanks.

Reply:
Ok, I'll check it. Thanks!